fili-core/src/main/java/com/yahoo/bard/webservice/data/DruidResponseParser.java

```diff
@@ -38,20 +39,28 @@
      *
      * @param jsonResult  Druid results in json
      * @param schema  Schema for results
-     * @param queryType  the type of query
+     * @param queryType  the type of query, note that this implementation only supports instances of
+     * {@link DefaultQueryType}
      *
      * @return the set of results
      */
     public ResultSet parse(JsonNode jsonResult, ZonedSchema schema, QueryType queryType) {
 
         LOG.trace("Parsing druid query {} by json result: {} using schema: {}", queryType, jsonResult, schema);
 
+
+        if (!(queryType instanceof DefaultQueryType)) {
+            // Throw an exception for unsupported query types
+            unsupportedQueryType(queryType);
+        }
+        DefaultQueryType defaultQueryType = (DefaultQueryType) queryType;
+
         /* Get dimension and metric columns */
         Set<DimensionColumn> dimensionColumns = schema.getColumns(DimensionColumn.class);
         Set<MetricColumn> metricColumns = schema.getColumns(MetricColumn.class);
 
-        List<Result> results;
-        switch (queryType) {
+        List<Result> results = null;
```

Comment:
It's needed to make the java compiler happy. Otherwise, it complains that `results` may not have been initialized (likely because it doesn't realize that the default case throws an exception, now that the exception throwing is hidden behind a method.